toolkit/offchain/src/reserve/release.rs

```diff
@@ -106,7 +106,13 @@ fn reserve_release_tx(
 	let mut tx_builder = TransactionBuilder::new(&get_builder_config(ctx)?);
 
 	let reserve_balance = previous_reserve.utxo.get_asset_amount(&token);
-	let amount_to_transfer = cumulative_total_transfer - stats.token_total_amount_transferred;
+	let token_total_amount_transferred = stats.token_total_amount_transferred;
+	if cumulative_total_transfer <= token_total_amount_transferred {
+		return Err(anyhow!(
+			"The requested total amount: {cumulative_total_transfer} is not greater than the amount already transferred to illiquid supply: {token_total_amount_transferred}.",
+		));
+	}
+	let amount_to_transfer = cumulative_total_transfer - token_total_amount_transferred;
```

Comment:
I'm wondering about moving these calculation out of this function, and passing `amount_to_transfer` and `left_in_reserve`. This would allow logging `amount_to_transfer` to user one time. OTOH this need should go away if we fix UI.

Reply:
what about somethnig like this?
```rust
	let amount_to_transfer = cumulative_total_transfer
		.checked_sub(token_total_amount_transferred)
		.ok_or(anyhow!("Overflow"))?;
```

Reply:
Yes, more idiomatic. But the question about place where to check user input remains. I tend to go near the edge.
Pros for having it here is that it is easier to test, because no mocks are involved.

Reply:
If there will be an option to check it earlier that would be great, but we should be also careful to not have a check in one place of the codebase and computations that make assumptions on top of it in the other. If we will not eliminate a subtraction, then keeping additional check here is a good idea

Reply:
I meant to pass result of subtraction to this function